src/hotspot/share/cds/filemap.cpp

```diff
@@ -842,6 +885,68 @@ bool FileMapInfo::classpath_failure(const char* msg, const char* name) {
   return false;
 }
 
+char* FileMapInfo::longest_common_app_classpath(int num_paths,
+                                                GrowableArray<const char*>* rp_array) {
+  // find the shortest path length among all the entries
+  char* chr = strrchr((char*)rp_array->at(0), *os::file_separator());
+  size_t min_len = chr != NULL ? chr - rp_array->at(0) + 1 : strlen(rp_array->at(0)) + 1;
+  for (int i = 1; i < num_paths; i++) {
+    chr = strrchr((char*)rp_array->at(i), *os::file_separator());
+    size_t curr_len = chr != NULL ? chr - rp_array->at(i) + 1 : strlen(rp_array->at(i)) + 1;
+    min_len = MIN2(min_len, curr_len);
+  }
+
+  // allocate a buffer to store the longest common prefix
+  char* lcp = NEW_C_HEAP_ARRAY(char, min_len + 1, mtInternal);
+  memset(lcp, 0, min_len + 1);
+  char c;
+  // find the longest common prefix
+  for (int j = 0; j < (int)min_len; j++) {
+    c = rp_array->at(0)[j];
+    int k = 1;
+    while (k < num_paths) {
+      if (rp_array->at(k)[j] != c) {
+        break;
+      }
+      k++;
+    }
+    *(lcp + j) = c;
+  }
+
+  if (*lcp == '\0') {
+    // no longest common prefix, release the buffer
+    FREE_C_HEAP_ARRAY(char, lcp);
+    return NULL;
+  } else {
+    return lcp;
+  }
+}
+
+bool FileMapInfo::check_paths_ignoring_common_path(int shared_path_start_idx, int num_paths,
+                                                   GrowableArray<const char*>* rp_array,
+                                                   const char* dumptime_prefix, const char* runtime_prefix) {
+  int i = 0;
+  int j = shared_path_start_idx;
+  bool mismatch = false;
+  while (i < num_paths && !mismatch) {
+    while (shared_path(j)->from_class_path_attr()) {
+      // shared_path(j) was expanded from the JAR file attribute "Class-Path:"
+      // during dump time. It's not included in the -classpath VM argument.
+      j++;
+    }
+    assert(strlen(shared_path(j)->name()) > strlen(dumptime_prefix), "sanity");
+    const char* dumptime_path = shared_path(j)->name() + strlen(dumptime_prefix);
+    assert(strlen(rp_array->at(i)) > strlen(runtime_prefix), "sanity");
+    const char* runtime_path = rp_array->at(i)  + strlen(runtime_prefix);
+    if (strcmp(dumptime_path, runtime_path) != 0) {
+      mismatch = true;
```

Comment:
Can this be simplified as `return false` so you don't need to maintain the `mismatch` variable?

Reply:
Fixed.